admin/cleanup.py

```diff
@@ -230,22 +251,107 @@ def _filter_test_volumes(self, maximum_age, volumes):
                 keep.append(volume)
         return VolumeActions(destroy=destroy, keep=keep)
 
-    def _destroy_cloud_volumes(self, volumes):
-        """
-        Unconditionally and irrevocably destroy all of the given cloud volumes.
-        """
-        for volume in volumes:
-            try:
-                volume.destroy()
-            except:
-                err(None, "Destroying volume.")
+
+@attributes(['lag', 'prefixes'])
+class CleanAcceptanceNodes(object):
+    """
+    :ivar timedelta lag: The age of nodes to destroy.
+    :param prefixes: List of prefixes of nodes to destroy.
+    """
+    def start(self, config):
+        # Get the libcloud drivers corresponding to the acceptance tests.
+        rackspace = get_rackspace_driver(config["rackspace"])
+        ec2 = get_ec2_driver(config["aws"])
+        drivers = [rackspace, ec2]
+
+        # Get the prefixes of the node names, appending the creator from
+        # the config.
+        creator = config['metadata']['creator']
+        prefixes = tuple(map(lambda prefix: prefix + creator, self.prefixes))
```

Comment:
:question: It'd be cool if this could somehow re-use the code from https://github.com/ClusterHQ/flocker/blob/master/admin/acceptance.py#L767. In general, I just have some concern about the fact that if I change that code, I wouldn't necessarily think to change this code.

That said, that's probably some sort of ideal world that we don't need to live in.